### DISCUSSION TITLE
Question on how to achieve a few effects (more like Corfu)

### DISCUSSION THREAD

**Discussion by KaranAhlawat** [company-mode/company-mode#1416] [Q&A] [answered]
Hi, I'd like to achieve the following in company-mode, but wasn't really able to figure out how. 

1. Previewing the current candidate in the buffer (this is not super important, as I understand `tng-mode` interferes a lot with LSP stuff)
2. If there is only a single candidate, I'd still like to have the tooltip menu rather than it being in my buffer (habit/preference coming from Corfu)

Any help is appreciated :)

**Comment by dgutov** [comment, marked as answer]
1. Depends on what you mean by "previewing". If you like how LSP returns completions (with args), you could replace `company-preview-if-just-one-frontend` with `company-preview` in `company-frontends`, then the "sole-completion" preview will be shown always.
2. And this one's easy: replace `company-preview-if-just-one-frontend` with `company-preview-frontend` by customizing the same option: `company-frontends`.

**Reply by KaranAhlawat** [reply]
Thank you, I ended up going with this

```lisp
(setq company-frontends '( company-pseudo-tooltip-frontend
                           company-preview-frontend
                           company-echo-metadata-frontend ))
```

**Reply by dgutov** [reply]
Welcome!

BTW, any chance for a better-searchable question title? The second question is for "like Corfu", or "like VS Code", I suppose. Was the first one inspired by some UI as well?

**Reply by KaranAhlawat** [reply]
Well they were both inspired by Corfu now that I think about it. We can change it to something like "Configuring company to be behave more like corfu" or something?

**Reply by dgutov** [reply]
Very good. I've updated the title, thanks!

**Reply by KaranAhlawat** [reply]
You guys have been super helpful for the past couple days, thanks alot 😄
